javascript/script.js

```diff
@@ -0,0 +1,185 @@
+
+  function randomLowerCaseCharacter() {
+    const lowerCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const lowerCaseArray = lowerCaseCharacters.split("");
+    const generateRandomLowerCC =
+      lowerCaseArray[Math.floor(Math.random() * lowerCaseArray.length)];
+    return generateRandomLowerCC;
+  }
+
+  function randomUppercaseCharacter() {
+    const upperCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const upperCaseArray = upperCaseCharacters.toUpperCase().split("");
+    const generateRandomUpperCC =
+      upperCaseArray[Math.floor(Math.random() * upperCaseArray.length)];
+    return generateRandomUpperCC;
+  }
+
+  function randomNumericalValue() {
+    const numericalValue = "0123456789";
+    const numericalArray = numericalValue.split("");
+    const generateRandomNumericalValue =
+      numericalArray[Math.floor(Math.random() * numericalArray.length)];
+    return generateRandomNumericalValue;
+  }
+
+  function randomSpecialCharacter() {
+    const specialCharacters = "@|!#$%&/()=?»«@£§€{}.-;<>_,";
+    const specialCharactersArray = specialCharacters.split("");
+    const generateRandomSpecialCharacter =
+      specialCharactersArray[
+        Math.floor(Math.random() * specialCharactersArray.length)
+      ];
+    return generateRandomSpecialCharacter;
+  }
+
+  //                       STEP 8  :
+  // store the above functions in a variable
+
+  const passwordLength = userPasswordLength();
+
+  const inputLowerCase = lowerCaseOption();
+
+  const inputUpperCase = upperCaseOption();
+
+  const inputNumericValue = numericOption();
+
+  const inputSpecialCharacter = specialCharacterOption();
+
+  //                       STEP 9  :
+
+  function generatingUserPassword() {
+    let randomGeneratedPassword = [];
+
+    while (passwordLength > randomGeneratedPassword.length) {
+      if (
+        inputLowerCase === false &&
+        inputUpperCase === false &&
+        inputNumericValue === false &&
+        inputSpecialCharacter === false
+      ) {
+        alert(
+          "Please pick one of the choice 1 type of character format for your password"
+        );
+        break;
+      }
+      if (inputLowerCase) {
+        randomGeneratedPassword.push(randomLowerCaseCharacter());
+      }
+
+      if (inputUpperCase) {
+        randomGeneratedPassword.push(randomUppercaseCharacter());
+      }
+
+      if (inputNumericValue) {
+        randomGeneratedPassword.push(randomNumericalValue());
+      }
+
+      if (inputSpecialCharacter) {
+        randomGeneratedPassword.push(randomSpecialCharacter());
+      }
+    }
+
+    while (randomGeneratedPassword.length - passwordLength > 0) {
+      randomGeneratedPassword.pop();
+    }
+
+    randomGeneratedPassword = randomGeneratedPassword.sort(
+      () => Math.random() - 0.5
+    );
+
+    randomGeneratedPassword = randomGeneratedPassword.join("");
+
+    return randomGeneratedPassword;
+  }
+  return generatingUserPassword();
```

Comment:
You are trying to return the function you are currently inside of - what is the purpose of this?